servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -2603,6 +2603,55 @@ public Template getTemplate(ResourceContext ctx, String templateName) {
         return template;
     }
 
+    @Override
+    public DomainTemplateDetailsList getDomainTemplateDetailsList(ResourceContext ctx, String domainName) {
+        DomainTemplateDetailsList domainTemplateDetailsList = new DomainTemplateDetailsList();
+        List<TemplateMetaData> templateMetaDataList = new ArrayList<>();
+        Map<String, Integer> templateDomainMapping = new HashMap<>();
+        final String caller = "getDomainTemplateDetailsList";
+        metric.increment(ZMSConsts.HTTP_GET);
+        metric.increment(ZMSConsts.HTTP_REQUEST);
+        metric.increment(caller);
+        logPrincipal(ctx);
+        validateRequest(ctx.request(), caller);
+        validate(domainName, TYPE_DOMAIN_NAME, caller);
+
+        // for consistent handling of all requests, we're going to convert
+        // all domain into lower case
+
+        domainName = domainName.toLowerCase();
+        final String principalDomain = getPrincipalDomain(ctx);
+        Object timerMetric = metric.startTiming("getdomainrolemembers_timing", domainName, principalDomain);
+
+        templateDomainMapping = dbService.getDomainTemplates(domainName);
+        if (templateDomainMapping != null) {
+            try {
+                for (String templateName : templateDomainMapping.keySet()) {
+                    TemplateMetaData metaData = new TemplateMetaData();
+                    Template template = serverSolutionTemplates.get(templateName);
+                    // there is a possibility of a stale template coming back from DB over time(caused by template clean up)
+                    if (template != null) {
+                        //Merging template metadata fields from solution-templates.json and template data from DB
+                        metaData.setTemplateName(templateName);
+                        metaData.setCurrentVersion(templateDomainMapping.get(templateName));
+                        metaData.setLatestVersion(template.getMetadata().getLatestVersion());
+                        metaData.setAutoUpdate(template.getMetadata().getAutoUpdate());
+                        metaData.setDescription(template.getMetadata().getDescription());
+                        metaData.setKeywordsToReplace(template.metadata.getKeywordsToReplace());
+                        metaData.setTimestamp(template.metadata.getTimestamp());
+                        templateMetaDataList.add(metaData);
+                    }
+                }
+                domainTemplateDetailsList.setMetaData(templateMetaDataList);
+            } catch (Exception ex) {
```

Comment:
what are the possible cases for exception and why are hiding them?

otherwise it looks like if there is some type of exception we're going to return an empty list to the client which is not correct.

so unless we're absolutely sure we want to swallow the exception, we shouldn't catch it and let an error be returned to the client

Reply:
there is absolutely no situation where this code will throw an exception that needs to be handled because it is mostly local object sets and list/map functions. even if it is throwing an error it should be thrown back to the client YES!. i added it while simulating a test scenario in dev box and did not remove. i will take care of this

Reply:
Checked in updated changes